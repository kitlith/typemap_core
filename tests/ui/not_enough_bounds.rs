use typemap_core::{typemap, TypeMapGet};

fn not_enough_bounds<Opts: TypeMapGet>(opts: &Opts) {
    println!("{}", opts.get::<u32>());
}

fn main() {
    not_enough_bounds(&typemap!(u32 = 42u32));
}
