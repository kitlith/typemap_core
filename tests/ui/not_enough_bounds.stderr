error[E0277]: the trait bound `Opts: Contains<u32>` is not satisfied
 --> $DIR/not_enough_bounds.rs:4:25
  |
4 |     println!("{}", opts.get::<u32>());
  |                         ^^^ the trait `Contains<u32>` is not implemented for `Opts`
  |
help: consider further restricting this bound
  |
3 | fn not_enough_bounds<Opts: TypeMapGet + Contains<u32>>(opts: &Opts) {
  |                                       ^^^^^^^^^^^^^^^
